packages/context/src/resolver.ts

```diff
@@ -129,6 +136,31 @@ export function resolveInjectedArguments(
   }
 }
 
+/**
+ * Invoke an instance method with dependency injection
+ * @param target The instance or class prototype
+ * @param method Name of the method
+ * @param ctx Context
+ * @param instance An instance of the class
+ */
+export function invokeMethod<T>(
+  // tslint:disable-next-line:no-any
+  target: any,
+  method: string,
+  ctx: Context,
+): ValueOrPromise<BoundValue> {
+  const argsOrPromise = resolveInjectedArguments(target, ctx, method);
+  let result: ValueOrPromise<BoundValue>;
+  if (isPromise(argsOrPromise)) {
+    // Instantiate the class asynchronously
+    result = argsOrPromise.then(args => target[method](...args));
```

Comment:
Can we simply `return argsOrPromise.then(...)` please, instead of using a mutable `result` variable?

Reply:
+1